src/components/AugmentedResultList/AugmentedResultList.ts

```diff
@@ -0,0 +1,142 @@
+import { $$, ComponentOptions, IComponentBindings, Initialization, IQueryResult, IResultListOptions, IFieldOption } from 'coveo-search-ui';
+
+/**
+ * Interface for the object data returned from remote action.
+ *
+ * @export
+ * @interface IObjectData
+ */
+export interface IObjectData {
+    /**
+     * Data specific to a result with matching object id.
+     */
+    objectData: {}[];
+    /**
+     * Data to add to every result with matching object id.
+     */
+    commonData: {};
+}
+
+/**
+ * Generic interface for the response returned by the remote action method.
+ *
+ * @export
+ * @interface IPromiseReturnArgs
+ * @template T
+ */
+export interface IPromiseReturnArgs {
+    data: IObjectData;
+}
+
+export interface AugmentedResultListOptions extends IResultListOptions {
+    matchingIdField: IFieldOption;
+    objectDataAction: (objectIds: String[]) => Promise<IPromiseReturnArgs>;
+}
+
+export class AugmentedResultList extends Coveo.ResultList implements IComponentBindings {
+    static ID = 'AugmentedResultList';
+
+    /**
+     * @componentOptions
+     */
+    static options: AugmentedResultListOptions = {
+        /**
+         * The field to be used as matching ID between object and result.
+         */
+        matchingIdField: ComponentOptions.buildFieldOption({
```

Comment:
In my local branch here i have a warning message that logs when the field value is replaced. This should make the cause fairly easy to spot if a problem arises. To have a generic component like this I think it needs to be fairly loose. So the responsibility would be on the documentation and the user of the component to exercise caution. Do you have another idea?

Reply:
Having a comparision method as a parameter instead ? This will allow more complex cases and it's still simple enough to easily cover the base schenario.

Reply:
Not a bad idea. i'll test it out